invokeai/frontend/web/src/features/settingsAccordions/components/AdvancedSettingsAccordion/AdvancedSettingsAccordion.tsx

```diff
@@ -63,11 +63,17 @@ export const AdvancedSettingsAccordion = memo(() => {
   );
   const badges = useAppSelector(selectBadges);
   const { t } = useTranslation();
-  const { isOpen, onToggle } = useStandaloneAccordionToggle({
+  const { isOpen, onToggle, onClose } = useStandaloneAccordionToggle({
     id: 'advanced-settings',
     defaultIsOpen: false,
   });
 
+  useEffect(() => {
+    if (activeTabName === 'upscaling') {
+      onClose();
+    }
+  }, [activeTabName, onClose]);
+
```

Comment:
Not a fan of `useEffect` for stuff like this. Also, this change means if you switch from generation tab to upscale and back, your opened accordions are now closed.

How about this, which remembers each tab's accordion state independently:
```suggestion
  const { isOpen, onToggle, onClose } = useStandaloneAccordionToggle({
    id: `advanced-settings-${activeTabName}`,
    defaultIsOpen: false,
  });
```

Same thing for the other accordion.